src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -5822,6 +5864,316 @@ .maxstack  2
+                      .maxstack  2
+                      .locals init (__InlineArray3<int?> V_0)
+                      IL_0000:  ldloca.s   V_0
+                      IL_0002:  initobj    "__InlineArray3<int?>"
+                      IL_0008:  ldloca.s   V_0
+                      IL_000a:  ldc.i4.0
+                      IL_000b:  call       "InlineArrayElementRef<__InlineArray3<int?>, int?>(ref __InlineArray3<int?>, int)"
+                      IL_0010:  ldc.i4.1
+                      IL_0011:  newobj     "int?..ctor(int)"
+                      IL_0016:  stobj      "int?"
+                      IL_001b:  ldloca.s   V_0
+                      IL_001d:  ldc.i4.1
+                      IL_001e:  call       "InlineArrayElementRef<__InlineArray3<int?>, int?>(ref __InlineArray3<int?>, int)"
+                      IL_0023:  ldc.i4.2
+                      IL_0024:  newobj     "int?..ctor(int)"
+                      IL_0029:  stobj      "int?"
+                      IL_002e:  ldloca.s   V_0
+                      IL_0030:  ldc.i4.2
+                      IL_0031:  call       "InlineArrayElementRef<__InlineArray3<int?>, int?>(ref __InlineArray3<int?>, int)"
+                      IL_0036:  initobj    "int?"
+                      IL_003c:  ldloca.s   V_0
+                      IL_003e:  ldc.i4.3
+                      IL_003f:  call       "InlineArrayAsSpan<__InlineArray3<int?>, int?>(ref __InlineArray3<int?>, int)"
+                      IL_0044:  call       "System.ReadOnlySpan<int?> System.Span<int?>.op_Implicit(System.Span<int?>)"
+                      IL_0049:  call       "MyCollection<int?> MyCollectionBuilder.Create<int?>(System.ReadOnlySpan<int?>)"
+                      IL_004e:  ret
+                    }
+                    """);
+            }
+            else
+            {
+                verifier.VerifyIL("Program.F",
+                    """
+                    {
+                      // Code size       43 (0x2b)
+                      .maxstack  4
+                      IL_0000:  ldc.i4.3
+                      IL_0001:  newarr     "int?"
+                      IL_0006:  dup
+                      IL_0007:  ldc.i4.0
+                      IL_0008:  ldc.i4.1
+                      IL_0009:  newobj     "int?..ctor(int)"
+                      IL_000e:  stelem     "int?"
+                      IL_0013:  dup
+                      IL_0014:  ldc.i4.1
+                      IL_0015:  ldc.i4.2
+                      IL_0016:  newobj     "int?..ctor(int)"
+                      IL_001b:  stelem     "int?"
+                      IL_0020:  newobj     "System.ReadOnlySpan<int?>..ctor(int?[])"
+                      IL_0025:  call       "MyCollection<int?> MyCollectionBuilder.Create<int?>(System.ReadOnlySpan<int?>)"
+                      IL_002a:  ret
+                    }
+                    """);
+            }
+        }
+
+        // As above, but with TargetFramework.NetFramework.
+        [ConditionalFact(typeof(DesktopOnly))]
```

Comment:
Why is this `DesktopOnly`? The tests doesn't execute, just compiles and verifies so that should work across all runtimes (unless I'm missing something).

Reply:
The test uses `CreateCompilationWithSpanAndMemoryExtensions()` which has different behavior on desktop and CoreCLR. Desktop is the more important case here since we're interested in testing .NET Framework, so I left the test as desktop only.